rocketpy/simulation/flight.py

```diff
@@ -1053,6 +726,573 @@ def __simulate(self, verbose):
+                gravity=self.env.gravity.get_value_opt(
+                    y_sol[2] if len(y_sol) > 2 else self.solution[-1][3]
+                ),
+                pressure=self.env.pressure,
+                earth_radius=self.env.earth_radius,
+                initial_coordinates=(self.env.latitude, self.env.longitude),
+            )
+
+    def __check_and_handle_parachute_triggers(
+        self, node, phase, phase_index, node_index
+    ):
+        """Check for parachute triggers and handle deployment.
+
+        Parameters
+        ----------
+        node : TimeNode
+            The current time node.
+        phase : FlightPhase
+            The current flight phase.
+        phase_index : int
+            The index of the current phase.
+        node_index : int
+            The index of the current node.
+
+        Returns
+        -------
+        bool
+            True if a parachute was triggered and the phase should break.
+        """
+        for parachute in node.parachutes:
+            # Calculate and save pressure signal
+            (
+                noisy_pressure,
+                height_above_ground_level,
+            ) = self.__calculate_and_save_pressure_signals(
+                parachute, node.t, self.y_sol[2]
+            )
+            if not parachute.triggerfunc(
+                noisy_pressure,
+                height_above_ground_level,
+                self.y_sol,
+                self.sensors,
+            ):
+                return False  # Early exit: parachute not deployed
+
+            # Remove parachute from flight parachutes
+            self.parachutes.remove(parachute)
+
+            # Create phase for time after detection and before inflation
+            # Must only be created if parachute has any lag
+            i = 1
+            if parachute.lag != 0:
+                self.flight_phases.add_phase(
+                    node.t,
+                    phase.derivative,
+                    clear=True,
+                    index=phase_index + i,
+                )
+                i += 1
+
+            # Create flight phase for time after inflation
+            callbacks = [
+                lambda self, parachute_cd_s=parachute.cd_s: setattr(
+                    self, "parachute_cd_s", parachute_cd_s
+                )
+            ]
+            self.flight_phases.add_phase(
+                node.t + parachute.lag,
+                self.u_dot_parachute,
+                callbacks,
+                clear=False,
+                index=phase_index + i,
+            )
+
+            # Prepare to leave loops and start new flight phase
+            phase.time_nodes.flush_after(node_index)
+            phase.time_nodes.add_node(self.t, [], [], [])
+            phase.solver.status = "finished"
+            self.parachute_events.append([self.t, parachute])
+            return True
+
+        return False
+
+    def __check_simulation_events(self, phase, phase_index, node_index):
```

Comment:
Since the three methods inside it are handlers, I would also call this one a handler.